### PR TITLE
Fixes #27196 - Add roles_path to ansible.cfg

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ares** [comment]
LGTM, could someone please take a look? it blocks https://github.com/theforeman/smart_proxy_ansible/pull/19 and is a high prio for us

File: manifests/plugin/ansible.pp [hunk 1]
```diff
@@ -22,13 +22,17 @@
...
+#               Can contain multiple paths separated by :
+#
 class foreman_proxy::plugin::ansible (
   Boolean $enabled = $::foreman_proxy::plugin::ansible::params::enabled,
   Foreman_proxy::ListenOn $listen_on = $::foreman_proxy::plugin::ansible::params::listen_on,
   Stdlib::Absolutepath $ansible_dir = $::foreman_proxy::plugin::ansible::params::ansible_dir,
   Optional[Stdlib::Absolutepath] $working_dir = $::foreman_proxy::plugin::ansible::params::working_dir,
   Boolean $host_key_checking = $::foreman_proxy::plugin::ansible::params::host_key_checking,
   String $stdout_callback = $::foreman_proxy::plugin::ansible::params::stdout_callback,
+  String $roles_path = $::foreman_proxy::plugin::ansible::params::roles_path,
```
Comment (ekohl) [inline manifests/plugin/ansible.pp]: Should this be a `Array[Stdlib::Absolutepath]` and then joined in the template?

File: templates/plugin/ansible.yml.erb [hunk 2]
```diff
@@ -6,3 +6,4 @@
 <% else -%>
 :working_dir: <%= scope.lookupvar("foreman_proxy::plugin::ansible::working_dir") %>
 <% end -%>
+:ansible_roles_path: <%= scope.lookupvar("foreman_proxy::plugin::ansible::roles_path") %>
```
Comment (ekohl) [inline templates/plugin/ansible.yml.erb]: If you're always reading this fro the ansible.cfg and we ensure it's there, I would leave this out. It's only a risk of them getting out of sync.

Comment (xprazak2) [reply templates/plugin/ansible.yml.erb]: Don't both of them use the same variable? But I see your point about duplicity, removed.

**Comment by ekohl** [comment]
If you see this being cherry picked then I'd appreciate a Redmine issue.

**Comment by xprazak2** [comment]
I added a redmine issue.

**Comment by ekohl** [comment]
Thanks!

**Comment by tbrisker** [comment]
looks like nightly installer build is now [failing](http://koji.katello.org/koji/taskinfo?taskID=208415) with:
```
 Unable to continue because of: undocumented parameters in foreman_proxy_plugin_ansible: ansible_dir, enabled, host_key_checking, listen_on, roles_path, stdout_callback, working_dir
```

File: manifests/plugin/ansible.pp [hunk 3]
```diff
@@ -21,6 +21,9 @@
 #                      environments like Foreman Remote Execution or cron
 #
 # $stdout_callback:: Ansible's stdout_callback setting
+#
+# $roles_path:: Paths where we look for ansible roles.
+
```
Comment (ekohl) [inline manifests/plugin/ansible.pp]: I think this empty line may have broken things. Oddly enough our linter is flexible enough that it didn't catch it but puppet-strings is a bit more strict

**Pull request merged**